pkg/sidecarlogresults/sidecarlogresults_test.go

```diff
@@ -0,0 +1,195 @@
+package sidecarlogresults
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/test/diff"
+)
+
+func TestLookForResults(t *testing.T) {
+	for _, c := range []struct {
+		desc              string
+		resultName        string
+		resultValue       string
+		generateResultDir bool
+		generateResult    bool
+		want              string
```

Comment:
Try to check error types where possible, not error strings. (i.e. https://pkg.go.dev/os#IsNotExist)